drools-parser/src/test/java/org/drools/parser/MiscDRLParserTest.java

```diff
@@ -624,7 +659,7 @@ void parse_lhsWithStringQuotesEscapeChars() {
 
     @Test
     void parse_literalBoolAndNegativeNumbersRule() throws Exception {
-        RuleDescr rule = parseAndGetFirstRuleFromFile("literal_bool_and_negative.drl");
+        RuleDescr rule = parseAndGetFirstRuleDescrFromFile("literal_bool_and_negative.drl");
```

Comment:
HI @tkobayas Thanks!
IIUC, for all the "drl" files the `parseAndGetFirstRuleDescrFromFile` method is invoked. This method is a unit test by itself (it has a couple of assertions). I was wondering if it would not be better to 
1. make it a "proper" test (`@Test`)
2. make it a parameterized test  with all the files name to iteratively invoke this method (i.e. for each file name in the parameter collection, this test is invoked)

Then, all the other tests would be specific for the file/condition that that test should actually verify: DTMS ?

Reply:
Thank you for the suggestion, @gitgabrio . You mean running 2 tests (A: parse and assert parse error. B: parse and drl specific test) per 1 file, correct? (even though A is a parameterized test)

I think the assertions in `parseAndGetFirstRuleDescrFromFile` are optional (= not the main purpose of the tests), so we may not need to make it a proper test (= running 2 tests for 1 file). But I may misunderstand. @pibizza @hellowdan Do you have any thoughts on the Gabriele's suggestion? If it improves the test quality or maintainability, I would accept.

Anyway, I filed https://issues.redhat.com/browse/DROOLS-7312 for refactoring MiscDRLParserTest to make it better.

Thanks again.

Reply:
Hi @tkobayas 
Yes, that was my suggestion. But 
`I think the assertions in parseAndGetFirstRuleDescrFromFile are optional (= not the main purpose of the tests), so we may not need to make it a proper test (= running 2 tests for 1 file)`
make sense, so let's deal with that as you think is better.